src/ringbuffer.toit

```diff
@@ -15,24 +15,39 @@ class RingBuffer:
   append value/float:
     buffer[head] = value
     head = (head + 1) % size_
-    count = min (count + 1) size_
+    count = (min (count + 1) size_)
 
+  is_empty -> bool:
+    return count == 0
+
+  reduce [block]:
+    if is_empty: throw "Not enough elements"
+    result := null
+    is_first := true
+    for i := 0; i < count; i++:
+      if is_first: result = buffer[i]; is_first = false
+      else: result = block.call result buffer[i]
+    return result
+
+  minimum -> float:
+    return reduce : | a b | min a b
+
+  maximum -> float:
+    return reduce : | a b | max a b
+
+  //@davidg238: what is this method for?
```

Comment:
I just added a minimum methode for some reason to check 2 values to which is the smaller one. Is there a different solution you would suggest? Couldn't find a system call (I also was not checking very hard)

Reply:
Yeah, they are tucked away in core/utils:  https://libs.toit.io/core/utils/library-summary#max(2%2C0%2C0%2C)